lib/mpl_toolkits/mplot3d/proj3d.py

```diff
@@ -130,20 +130,19 @@ def ortho_transformation(zfront, zback):
 
 def proj_transform_vec(vec, M):
     vecw = np.dot(M, vec)
-    w = vecw[3]
-    # clip here..
-    txs, tys, tzs = vecw[0]/w, vecw[1]/w, vecw[2]/w
-    return txs, tys, tzs
+    vecw /= vecw[3]
+    return vecw[0:3]
```

Comment:
Why the `0:3`?

Reply:
If I remove this tests fails, because I think `vecw.shape[0]` and `vec.shape[0]` is 4.

Should I change this to validation of `vec` shape before dot product with `M`?

Reply:
Yes, the lengths of those arrays is 4, this is expected for a transform operation. This can be rewritten as `return vecw[:3] / vecw[3]` It'll involve 25% fewer division operations.